corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -1264,15 +1262,15 @@ def _report_columns(self):
         for i, conf in enumerate(self.cleaned_data['columns']):
             columns.extend(
                 self.report_column_options[conf['property']].to_column_dicts(
-                    i, conf['display_text'], conf['calculation']
+                    i, conf['display_text'], conf['calculation'], is_aggregated_on=conf["property"] == self.aggregation_field
                 )
             )
 
         # Add the aggregation indicator to the columns if it's not already present.
-        displaying_agg_column = bool([c for c in columns if c['field'] == agg_field_id])
+        displaying_agg_column = bool([c for c in self.cleaned_data['columns'] if c['property'] == self.aggregation_field])
```

Comment:
(Kinda interesting but totally ignorable tip: If you use `any`  with a generator instead of `bool` with a list, then the expression will return on the first True instead of iterating all values. i.e.
```python
displaying_agg_column = any((c['property'] == self.aggregation_field for c in self.cleaned_data['columns']))
```
I wouldn't bother with a small number of list items like this, but it's a useful pattern when you have a long list.)